src/DynamoCoreWpf/ViewModels/Preview/WatchHoverIconViewModel.cs

```diff
@@ -0,0 +1,212 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+using System.Windows.Forms;
+using System.Windows.Threading;
+using Dynamo.Core;
+using Dynamo.Graph.Nodes;
+using Dynamo.Models;
+using Dynamo.ViewModels;
+using DSCore;
+using CoreNodeModels;
+using Dynamo.UI.Commands;
+
+namespace Dynamo.ViewModels
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class WatchHoverIconViewModel: NotificationObject
+    {
+        private ConnectorViewModel ViewModel { get; set; }
+        private DynamoViewModel DynamoViewModel { get; set; }
+        private Dispatcher Dispatcher { get; set; }
+
+        public Point MidPoint { get; private set; }
+        public double MarkerSize { get; private set; } = 30;
+
+        private bool _isHalftone;
+        public bool IsHalftone 
+        {
+            get
+            {
+                return _isHalftone;
+            }
+            set
+            {
+                _isHalftone = value;
+                RaisePropertyChanged(nameof(IsHalftone));
+            }
+        }
+
+        public DelegateCommand PlaceWatchNodeCommand { get; set; }
+
+        private void PlaceWatchNodeCommandExecute(object param)
+        {
+            ViewModel.DiscardAllConnectorPins();
+            PlaceWatchNode();
+        }
+
+        bool CanExecute(object parameter)
+        {
+            return true;
+        }
+
+        void InitCommands()
+        {
+            PlaceWatchNodeCommand = new DelegateCommand(PlaceWatchNodeCommandExecute,CanExecute);
+        }
+
+        public WatchHoverIconViewModel(ConnectorViewModel connectorViewModel, DynamoViewModel dynamoViewModel)
+        {
+            ViewModel = connectorViewModel;
+            DynamoViewModel = dynamoViewModel;
```

Comment:
Looks like you are only using the `Model` from the `DynamoViewModel` so could we not just pass the model as the arg?

Reply:
Done.